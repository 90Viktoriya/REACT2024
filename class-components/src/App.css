
:root {
  --purple: #3A2149;
  --dark: #280D18;
  --light: #F7F1D1;
  --blue: #496271;
  --brown: #DEBB66;
}
#root {
  width: 100%;
  min-height: 100svh;
  background-image: url(./assets/background.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-color: var(--dark);
  color: var(--light);
  text-align: center;
}


