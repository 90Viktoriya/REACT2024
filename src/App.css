
:root {
  --purple: #3A2149;
  --dark: #280D18;
  --light: #F7F1D1;
  --light-transparent: #F7F1D180;
  --blue: #496271;
  --brown: #DEBB66;
  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
  line-height: 1.5;
  font-weight: 400;
}
body {
  margin: 0;
  display: flex;
  place-items: center;
  min-width: 320px;
  min-height: 100vh;
}
#root {
  width: 100%;
  min-height: 100svh;
  box-sizing: border-box;
  background-image: url(./assets/background.png);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  background-color: var(--dark);
  color: var(--light);
  text-align: center;
}
p {
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
}
a {
  font-weight: 500;
  color: var(--blue);
  text-decoration: inherit;
}
h1 {
  font-size: 3.2em;
}

