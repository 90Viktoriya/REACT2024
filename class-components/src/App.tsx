import { Component } from 'react';
import { ErrorBoundary } from './components/ErrorBoundary';
import { ErrorThrow } from './features/ErrorThrowPart/ErrorThrowPart';
import { SearchPart } from './features/SearchPart/SearchPart';
import { MainPart } from './features/MainPart/MainPart';
import './App.css';
import { ComponentsCaptions } from './data/ComponentsCaptions';

export class App extends Component {
  state = { isLoaded: false };

  handleOnSearch = (isLoadedValue: boolean) => {
    this.setState({ isLoaded: isLoadedValue });
  };

  render() {
    return (
      <ErrorBoundary fallback={<h1>{ComponentsCaptions.ERROR_TITLE}</h1>}>
        <h1>Characters of Star Trek</h1>
        <ErrorThrow />
        <SearchPart onSearch={this.handleOnSearch} />
        <MainPart onSearch={this.handleOnSearch} isLoaded={this.state.isLoaded} />
      </ErrorBoundary>
    );
  }
}

export default App;
