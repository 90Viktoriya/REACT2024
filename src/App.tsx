import { ErrorBoundary } from './components/ErrorBoundary';
import './App.css';
import { ComponentsCaptions } from './data/ComponentsCaptions';
import { Router } from './features/Router/Router';

export function App() {
  return (
    <ErrorBoundary fallback={<h1>{ComponentsCaptions.ERROR_TITLE}</h1>}>
      <Router />
    </ErrorBoundary>
  );
}

export default App;
