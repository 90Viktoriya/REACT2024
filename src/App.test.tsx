import { describe, expect } from 'vitest';
import { render } from '@testing-library/react';
import App from '../pages/_app';
import { Router } from 'next/router';

describe('Given App component', () => {
  it('when rendered, should render', () => {
    const screen = render(<App Component={() => <div>Test</div>} pageProps={{}} router={{} as Router} />);

    expect(screen.queryByText('Test')).toBeTruthy();
  });
});
