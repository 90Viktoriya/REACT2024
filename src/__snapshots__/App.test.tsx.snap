// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given App component > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="_main_6d1f25"
  >
    <div
      class="_loader_bf668d"
    />
  </section>
</DocumentFragment>
`;
